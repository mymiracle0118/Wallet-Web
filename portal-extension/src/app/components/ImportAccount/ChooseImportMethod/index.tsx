import React from 'react'
import { useTranslation } from 'react-i18next'
import { CustomTypography } from '../../custom-typography'
import { IChooseImportMethodProps } from '@portal/shared/utils/types'
import { DocumentGradientIcon, KeyGradientIcon } from '../../Icons'

const ChooseWalletImportMethod = ({ importWallet, importByPrivateKey }: IChooseImportMethodProps) => {
  const { t } = useTranslation()

  // TODO :: NOTE temporary disabled when import after logged in , as functionality need to be implented.
  const myArray = window.location.href.split('/')
  const fromPage = myArray[myArray.length - 2]
  const disabled = fromPage === 'account' ? true : false

  return (
    <>
      <CustomTypography dataAid="recoveryQuestion" variant="h1">
        {t('Onboarding.chooseImortMethodTitle')}
      </CustomTypography>

      <div className="bg-surface-dark-alt cursor-pointer rounded-lg mt-8 p-4" onClick={disabled ? '' : importWallet}>
        <div className="flex items-start justify-between gap-x-4">
          <div>
            <CustomTypography dataAid="phraseGuide" variant="subtitle" className="pb-2">
              {t('Onboarding.bySecretRecoveryPhrase')}
            </CustomTypography>
            <CustomTypography dataAid="phraseGuide" variant="body" type="secondary">
              {t('Onboarding.toAddMultiChain')}
            </CustomTypography>
          </div>
          <div className="text-[1.5rem]">
            <DocumentGradientIcon />
          </div>
        </div>
      </div>
      <div className="bg-[#1B1C24] cursor-pointer rounded-lg mt-4 p-4" onClick={importByPrivateKey}>
        <div className="flex items-start justify-between space-x-4">
          <div>
            <CustomTypography dataAid="phraseGuide" variant="subtitle" className="pb-2">
              {t('Onboarding.byPrivateKey')}
            </CustomTypography>
            <CustomTypography dataAid="phraseGuide" variant="body" type="secondary">
              {t('Onboarding.toAddSingleChain')}
            </CustomTypography>
          </div>
          <div className="text-[1.5rem]">
            <KeyGradientIcon />
          </div>
        </div>
      </div>
    </>
  )
}

export default ChooseWalletImportMethod
