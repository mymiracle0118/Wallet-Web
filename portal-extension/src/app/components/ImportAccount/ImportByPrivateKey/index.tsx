import { Menu } from '@headlessui/react'
import { useSettings } from '@portal/shared/hooks/useSettings'
import { useWallet } from '@portal/shared/hooks/useWallet'
import { useNavigate } from 'lib/woozie'
import React, { Fragment, useCallback, useState } from 'react'
import { useTranslation } from 'react-i18next'
// import { default as assetsDefaultList } from '@portal/shared/data/assets.json'
import { yupResolver } from '@hookform/resolvers/yup'
import { Avatar } from '@nextui-org/react'
import { default as networkTokenList } from '@portal/shared/data/networkTokens.json'
import { NetworkFactory } from '@portal/shared/factory/network.factory'
import { useStore } from '@portal/shared/hooks/useStore'
import { ImportUsernameProps, NetworkToken } from '@portal/shared/utils/types'
import ChangeProfileAvatar from '@src/app/components/ChangeProfileAvatar'
import { SpinnerIcon } from '@src/app/components/Icons'
import { decryptData } from '@src/utils/constants'
import { Button, CustomTypography, Dropdown, DropdownItem, Form, Icon, Input } from 'app/components'
import PasswordPromptModal from 'app/pages/wallet/PasswordPromptModal'
import defaultAvatar from 'assets/images/Avatar.png'
import { ethers } from 'ethers'
import { useForm } from 'react-hook-form'
import * as yup from 'yup'
import DropDownIcon from '../../../../../public/images/backgrounds/dropdown_icon.svg'

const networkTokens = Object.values(networkTokenList).filter((token) => token.tokenType === 'Native')

const ImportByPrivateKeyComponents = ({ nextToFetchUsername }: ImportUsernameProps) => {
  const myArray = window.location.href.split('/')
  const fromPage = myArray[myArray.length - 2]

  const { t } = useTranslation()
  const { navigate } = useNavigate()
  const [privateKey, setPrivateKey] = useState<string>('')
  const [errorText, setErrorText] = useState<string>('')
  const [loading, setLoading] = useState<boolean>(false)
  const { clearWallet, storeWallet } = useWallet()
  const [activeNetwork, setActiveNetwork] = useState<NetworkToken>(networkTokens[0])
  const { clearAccounts, clearAddressbook, currentAccount } = useSettings()
  const [showUsernameView, setShowUsernameView] = useState(false)
  const { saveAccount } = useSettings()
  // const { importPrivateKey } = useWallet()
  const [openPasswordModal, setOpenPasswordModal] = useState<boolean>(false)
  const [selectedAvatar, setSelectedAvatar] = useState<null | string>(null)
  const [isShowAvatarModal, setShowAvatarModal] = useState<boolean>(false)
  const [username, setUsername] = useState('')
  const [responseData, setResponseData] = useState<any>()

  const { walletsList } = useStore()

  const schema = yup.object().shape({
    username: yup
      .string()
      .min(3, t('Account.usernameMinimum'))
      .max(15, t('Account.usernameMaximum'))
      .required(t('Account.usernameRequired'))
      .matches(/^\S*$/, t('Account.usernameNoSpace')),
  })

  const handleChangeNetwork = (val: string) => {
    const selected = networkTokens.find((v) => v.networkName === val) as NetworkToken
    setActiveNetwork(selected)
    setErrorText('')
  }

  const handleNextClick = useCallback(async () => {
    setLoading(true)
    try {
      if (privateKey && activeNetwork) {
        if (fromPage === 'account') {
          // import after login
          setLoading(false)
          const asset = networkTokens.find((v) => v.networkName === activeNetwork.networkName) as NetworkToken
          if (asset) {
            const networkFactory = NetworkFactory.selectByNetworkId(activeNetwork.networkName)
            const checkPKey = await networkFactory.checkAccountAlreadyImported(privateKey, activeNetwork.networkName)
            if (checkPKey.result) {
              setShowUsernameView(true)
            } else {
              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
              setErrorText(checkPKey.msg)
            }
          } else {
            setErrorText('Not supporting this network key!')
          }
        } else {
          // import when onboarding
          clearWallet()
          clearAccounts()
          clearAddressbook()

          const asset = networkTokens.find((v) => v.networkName === activeNetwork.networkName) as NetworkToken
          if (asset) {
            const networkFactory = NetworkFactory.selectByNetworkId(activeNetwork.networkName)
            const checkKey = await networkFactory.importWalletByPrivateKey(privateKey, activeNetwork.networkName)
            if (checkKey && checkKey.result) {
              const fetchusername = nextToFetchUsername ? nextToFetchUsername.toString() : '' // Use an empty string if nextToFetchUsername is undefined
              const chain = activeNetwork.networkName || '' // Use an empty string if activeNetwork.networkName is undefined
              navigate(fetchusername + '/' + chain)
            } else {
              setErrorText(checkKey.msg)
            }
            setLoading(false)
          } else {
            setErrorText('Somthing went wrong in network!')
            setLoading(false)
          }
        }
      }
    } catch (error) {
      setErrorText('Invalid key!')
      setLoading(false)
    }
  }, [activeNetwork, clearAccounts, clearAddressbook, clearWallet, navigate, privateKey])

  const methods = useForm({
    resolver: yupResolver(schema),
    mode: 'onChange',
  })
  const {
    register,
    handleSubmit,
    // setError,
    formState: { errors, isValid, isDirty },
    watch,
  } = methods

  const handleImportPrivateKey = ({ username }: { username: string }) => {
    setOpenPasswordModal(true)
    setUsername(username)
  }

  const handlePasswordPromptSuccess = async (password: string) => {
    try {
      setLoading(true)
      if (currentAccount) {
        const accountWallet = walletsList[currentAccount.address][currentAccount.networkName]
        const networkToken = useWallet.getState().getNetworkTokenWithCurrentEnv(currentAccount.networkName)
        let checkPassword: any
        if (networkToken.isEVMNetwork) {
          checkPassword = await ethers.Wallet.fromEncryptedJson(
            accountWallet.encryptedWallet as string,
            password as string
          )
        } else {
          checkPassword = await decryptData(accountWallet.encryptedWallet as string, password as string)
        }

        if (checkPassword) {
          const networkFactory = NetworkFactory.selectByNetworkId(activeNetwork.networkName)
          const { wallet, encryptedPrivateKey, address } = await networkFactory.importPrivateKey(privateKey)
          await storeWallet({ wallet, encryptedPrivateKey, address }, '', password, activeNetwork.networkName, address)
            .then(() => saveAccount(username, address, activeNetwork.networkName, false, true))
            .catch((e: Error) => console.log(e.message))
        }
        setResponseData({ status: true, data: {} })
        setOpenPasswordModal(false)
        setLoading(false)
        navigate('/account')
      }
    } catch (error) {
      setResponseData({ status: false, data: error })
      setLoading(false)
    }
  }

  const handlePasswordPromptFail = (error: never) => {
    console.log(error)
    setOpenPasswordModal(false)
    return error
  }

  return (
    <>
      {!showUsernameView && (
        <Fragment>
          <CustomTypography dataAid="addSingleWallets" variant="body">
            {t('Onboarding.toAddSingleChain')}
          </CustomTypography>

          <Dropdown
            classDynamicChild="h-[16.25rem] overflow-x-hidden overflow-y-scroll w-full border border-solid border-[#424250]"
            classDynamicMenu="bg-custom-white10 mb-4 mt-6 rounded-md p-1 !table"
            anchor={
              <Menu.Button data-aid="currencyDropdown" className="p-2 rounded-xl flex items-center gap-2">
                <img alt="icon" src={activeNetwork?.image} className="h-6 rounded-full" />

                <CustomTypography className="w-60 text-left mr-4" variant="subtitle">
                  {activeNetwork?.networkName}
                </CustomTypography>
                <Icon size="small" icon={<DropDownIcon />} />
              </Menu.Button>
            }
          >
            {networkTokens.map((network) => (
              <DropdownItem
                key={network.networkName}
                active={network.networkName === activeNetwork?.networkName}
                text={network.networkName}
                icon={network.image}
                isImg={true}
                onSelect={handleChangeNetwork}
              />
            ))}
          </Dropdown>
          <Input
            name="privateKey"
            multiline={3}
            dataTestId="input-private-key"
            placeholder="Enter your private key"
            value={privateKey}
            onChange={(event: React.ChangeEvent<HTMLTextAreaElement>) => {
              setPrivateKey(event.target.value), setErrorText('')
            }}
            mainColor
            // error={errors.privateKey?.message}
          />
          <span className="text-feedback-negative font-semibold mt-2">{errorText}</span>

          <Button
            type="submit"
            className="mt-4"
            data-aid="nextNavigation"
            data-test-id="button-next"
            onClick={handleNextClick}
            color={`${!privateKey.length || loading ? 'disabled' : 'primary'}`}
            isDisabled={!privateKey.length || loading}
          >
            {t('Actions.next')}
          </Button>
        </Fragment>
      )}
      {showUsernameView && (
        <Fragment>
          <Form methods={methods} onSubmit={handleSubmit(handleImportPrivateKey)}>
            <div className="space-y-4">
              <div className="flex flex-col items-center space-y-4">
                <Avatar src={selectedAvatar || defaultAvatar} className="w-24 h-24 rounded-full" />

                <Button
                  variant="bordered"
                  color="outlined"
                  className="font-extrabold h-11 w-24"
                  onClick={() => setShowAvatarModal(true)}
                >
                  {t('Actions.change')}
                </Button>
              </div>

              <div className="space-y-1">
                <CustomTypography variant="subtitle" type="secondary">
                  {t('Onboarding.accountName')}
                </CustomTypography>
                <Input
                  autoComplete="off"
                  dataTestId="username"
                  mainColor
                  id="username"
                  placeholder={t('Onboarding.accountName')}
                  endAdornment={
                    <div className="text-[12px] text-fotter-dark-inactive">{watch('username')?.length || 0}/15</div>
                  }
                  fullWidth
                  disabled={loading}
                  {...register('username')}
                  error={errors.username?.message}
                  dynamicColor={'text-feedback-negative'}
                />
              </div>
            </div>

            <div className="flex gap-2 justify-between mt-8">
              <Button
                type="submit"
                data-test-id="create-subaccount-btn"
                color={!isDirty || !isValid || loading ? 'disabled' : 'primary'}
                isDisabled={!isDirty || !isValid || loading}
                isLoading={loading}
                spinner={<SpinnerIcon />}
              >
                {!loading && t('Actions.import')}
              </Button>
            </div>

            <ChangeProfileAvatar
              openModal={isShowAvatarModal}
              closeModal={() => {
                setShowAvatarModal(false)
                setSelectedAvatar('')
              }}
              selectedAvatar={selectedAvatar}
              setSelectedAvatar={setSelectedAvatar}
              handleAvatarChange={() => setShowAvatarModal(false)}
            />
          </Form>
          {/* <PasswordPromptModal
            modalState={openPasswordModal}
            closePromptModal={() => setOpenPasswordModal(false)}
            onSuccess={handlePasswordPromptSuccess}
            getPassword={setPassword}
          /> */}
          <PasswordPromptModal
            modalState={openPasswordModal}
            closePromptModal={() => setOpenPasswordModal(false)}
            onPromptPassword={handlePasswordPromptSuccess}
            onFail={handlePasswordPromptFail}
            responseData={responseData}
            isDismissable={false}
            buttonDisable={loading}
          />
        </Fragment>
      )}
    </>
  )
}

export default ImportByPrivateKeyComponents
