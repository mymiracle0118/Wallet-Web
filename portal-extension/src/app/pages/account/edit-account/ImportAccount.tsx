import React, { useState, useCallback } from 'react'
import SinglePageTitleLayout from 'layouts/single-page-layout/SinglePageLayout'
import { useTranslation } from 'react-i18next'
import { Input, CustomTypography, PasswordInput, Form, Button } from 'app/components'
import { useNavigate, goBack } from 'lib/woozie'
import { useWallet } from '@portal/shared/hooks/useWallet'
import { useSettings } from '@portal/shared/hooks/useSettings'

import { useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import * as yup from 'yup'
import { QuestionMarkIcon, SpinnerIcon } from '@src/app/components/Icons'
import { Tooltip } from '@nextui-org/react'

const schema = yup.object().shape({
  privateKey: yup.string().required(),
  password: yup.string().min(8).max(32).required(),
})

const ImportAccount = () => {
  const { navigate } = useNavigate()
  const { t } = useTranslation()
  const [loading, setLoading] = useState<boolean>(false)
  const { openWallet, importPrivateKey } = useWallet()
  const { saveAccount } = useSettings()

  const methods = useForm({
    resolver: yupResolver(schema),
    mode: 'onChange',
  })
  const {
    handleSubmit,
    formState: { errors, isValid, isDirty },
    setError,
  } = methods

  const importAccountFromPrivateKey = useCallback(
    async (privateKey: string, password: string) => {
      saveAccount()
      await importPrivateKey(privateKey, password, 'mainnet', 'username')
      saveAccount()

      goBack()
    },
    [importPrivateKey, saveAccount]
  )

  const handleImportPrivateKey = useCallback(
    async (data: { privateKey: string; password: string }) => {
      setLoading(true)
      try {
        setLoading(true)
        await importAccountFromPrivateKey(data.privateKey, data.password)
        setLoading(false)
      } catch (error) {
        setLoading(false)
        setError('password', { type: 'custom', message: error.message })
      }
    },
    [openWallet, importAccountFromPrivateKey, setError]
  )

  return (
    <SinglePageTitleLayout title="Import account">
      <Form methods={methods} onSubmit={handleSubmit(handleImportPrivateKey)}>
        <div>
          <CustomTypography variant="body">{t('Account.importAccountMessage')}</CustomTypography>

          <div className="flex items-center justify-between my-4 gap-2">
            <Input fullWidth mainColor value="Private Key" placeholder="Private Key" disabled />
            <Tooltip content="Private Key" placement="top-end">
              <Button isIconOnly size="sm" variant="light" fullWidth={false}>
                <QuestionMarkIcon />
              </Button>
            </Tooltip>
          </div>
          <PasswordInput
            mainColor
            dataTestId="private-key-input"
            id="private-key"
            name="privateKey"
            placeholder="Enter your private key"
            error={errors.privateKey?.message}
            disabled={loading}
            className="mb-8"
          />
          <CustomTypography className="mt-2" type="secondary" variant="subtitle">
            {t('Account.enterPassword')}
          </CustomTypography>

          <PasswordInput
            mainColor
            id="enter-password"
            name="password"
            placeholder={t('Account.enterPassword')}
            error={errors.password?.message}
            subTitle={'Min 8 characters'}
            className="mb-8"
            disabled={loading}
          />
          <div className="flex mt-4 gap-2">
            <Button color="outlined" variant="bordered" onClick={() => navigate('/home')}>
              {t('Actions.cancel')}
            </Button>
            <Button
              data-test-id="import-privatekey-btn"
              color={`${!isValid || !isDirty || loading ? 'disabled' : 'primary'}`}
              isDisabled={!isValid || !isDirty || loading}
              type="submit"
              isLoading={loading}
              spinner={<SpinnerIcon />}
            >
              {!loading && t('Actions.import')}
            </Button>
          </div>
        </div>
      </Form>
    </SinglePageTitleLayout>
  )
}
export default ImportAccount
