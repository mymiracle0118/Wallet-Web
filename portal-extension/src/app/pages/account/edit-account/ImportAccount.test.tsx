/* eslint-disable @typescript-eslint/no-unsafe-call */

import React from 'react'
import userEvent from '@testing-library/user-event'
import { render, screen, renderHook, act } from '@testing-library/react'
import { useWallet } from '@portal/shared/hooks/useWallet'
import { useSettings } from '@portal/shared/hooks/useSettings'
import ImportAccount from 'pages/account/edit-account/ImportAccount'
import { I18nextProvider } from 'react-i18next'
import i18n from '@portal/shared/i18n/config'

describe('<ImportPrivateKey />', () => {
  test('renders correctly ImportPrivateKey Page', () => {
    const tree = render(<ImportAccount />)
    expect(tree).toMatchSnapshot()
  })

  test('Import Private key', async () => {
    const user = userEvent.setup()
    const { result: walletResult } = renderHook(() => useWallet())
    const { result: walletSettings } = renderHook(() => useSettings())

    act(() => {
      walletResult.current.address = '0xcea68AdfE6D57Ff021EE9E999364b3Fb3B99Ad8d'
      walletResult.current.encryptedWallet =
        '{"address":"cea68adfe6d57ff021ee9e999364b3fb3b99ad8d","id":"ac94190a-a0d0-4eee-a248-455853da00db","version":3,"Crypto":{"cipher":"aes-128-ctr","cipherparams":{"iv":"f787434fc33eeb392d6f21c7a0a6ff56"},"ciphertext":"1378d19477d26b7312f995bb8640d3f02cde6906950c4c984b756a552df3ffe6","kdf":"scrypt","kdfparams":{"salt":"0d53d628bca9550b5ff2aaebd0b78a41fdaaaa25f8c97700f32abf9c1c59552c","n":65536,"dklen":32,"p":1,"r":8},"mac":"41506475afdb6723d8f7c1487906af21fabffcaf16b08c1cd016274b1401cbd0"},"x-ethers":{"client":"ethers.js","gethFilename":"UTC--2022-07-26T05-09-34.0Z--cea68adfe6d57ff021ee9e999364b3fb3b99ad8d","mnemonicCounter":"ca31b478c7c0119d0167e44103165518","mnemonicCiphertext":"49d03b450a1b9105c767b2c5c8a33e30","path":"m/44\'/60\'/0\'/0/0","locale":"en","version":"0.1"}}'
      walletResult.current.lockTime = 5
      walletResult.current.lockWallet = false
      walletResult.current.username = 'Test 1'

      walletSettings.current.accounts = {
        '0xcea68AdfE6D57Ff021EE9E999364b3Fb3B99Ad8d': {
          address: '0xcea68AdfE6D57Ff021EE9E999364b3Fb3B99Ad8d',
          encryptedWallet:
            '{"address":"cea68adfe6d57ff021ee9e999364b3fb3b99ad8d","id":"ac94190a-a0d0-4eee-a248-455853da00db","version":3,"Crypto":{"cipher":"aes-128-ctr","cipherparams":{"iv":"f787434fc33eeb392d6f21c7a0a6ff56"},"ciphertext":"1378d19477d26b7312f995bb8640d3f02cde6906950c4c984b756a552df3ffe6","kdf":"scrypt","kdfparams":{"salt":"0d53d628bca9550b5ff2aaebd0b78a41fdaaaa25f8c97700f32abf9c1c59552c","n":65536,"dklen":32,"p":1,"r":8},"mac":"41506475afdb6723d8f7c1487906af21fabffcaf16b08c1cd016274b1401cbd0"},"x-ethers":{"client":"ethers.js","gethFilename":"UTC--2022-07-26T05-09-34.0Z--cea68adfe6d57ff021ee9e999364b3fb3b99ad8d","mnemonicCounter":"ca31b478c7c0119d0167e44103165518","mnemonicCiphertext":"49d03b450a1b9105c767b2c5c8a33e30","path":"m/44\'/60\'/0\'/0/0","locale":"en","version":"0.1"}}',
          username: 'Account 1',
        },
      }
    })

    render(
      <I18nextProvider i18n={i18n}>
        <ImportAccount />
      </I18nextProvider>
    )

    const privateKey = '93ffbdb1ce74b3832cee32444d3880f2f3ee1e457c27ee3b925c960827c7aa82'
    await user.type(screen.getByTestId('private-key-input'), privateKey)
    await user.type(screen.getByTestId('password'), '12341234')
    await user.click(screen.getByTestId('import-privatekey-btn'))
    await screen.findByTestId('progress-loading').then((el) => expect(el).toBeInTheDocument())
  }, 30000)
})
