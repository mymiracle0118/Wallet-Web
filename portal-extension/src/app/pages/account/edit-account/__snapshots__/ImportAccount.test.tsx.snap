// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportPrivateKey /> renders correctly ImportPrivateKey Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
      >
        <div
          class="flex flex-row relative justify-between items-center h-16"
        >
          <button
            class="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
            type="button"
          >
            <arrow-left.svg
              classname="text-xl stroke-custom-black dark:stroke-custom-white"
            />
          </button>
          <div
            class="flex-1 text-center"
          >
            <div
              class="text-custom-black dark:text-custom-white text-lg font-extrabold"
            >
              Import account
            </div>
          </div>
        </div>
        <div
          class="w-full h-full overflow-auto"
        >
          <form>
             
            <div
              class="css-1vn5bei"
            >
              <div
                class="text-custom-black dark:text-custom-white text-sm font-regular"
              >
                Account.importAccountMessage
              </div>
              <div
                class="MuiBox-root css-1jrf70h"
              >
                <div
                  class="flex relative rounded-md items-center hover:shadow-md w-full bg-input-light-bg dark:bg-custom-white10"
                >
                  <input
                    autocomplete="off"
                    class="outline-none	py-3.5 px-3.5 w-full text-sm text-custom-black dark:text-custom-white bg-transparent bg-transparent text-h4 "
                    disabled=""
                    name="test"
                    placeholder="Private Key"
                    value="Private Key"
                  />
                </div>
                <div
                  class="relative flex flex-col items-center group"
                >
                  <div
                    class="css-mzan16"
                  >
                    <svg
                      class="inline-block select-none shrink-0 m-auto fill-custom-black dark:fill-custom-white text-2xl"
                      style="height: 1em; width: 1em;"
                    >
                      <help-circle.svg />
                    </svg>
                  </div>
                  <div
                    class="absolute bottom-0 flex flex-col items-center hidden mb-6 group-hover:flex right-0 undefined"
                  >
                    <span
                      class="rounded-md dark:bg-surface-dark-alt bg-surface-light relative z-10 p-2 text-xs leading-none dark:text-custom-white text-custom-grey100 whitespace-no-wrap bg-black shadow-lg w-max max-w-[300px] mb-2"
                    >
                      Private Key
                    </span>
                    <div
                      class="w-3 h-3 -mt-2 rotate-45 bg-black"
                    />
                  </div>
                </div>
              </div>
              <div
                class="mt-4 mb-8"
              >
                <div
                  class="flex relative rounded-md items-center hover:shadow-md w-full bg-input-light-bg dark:bg-custom-white10"
                >
                  <input
                    autocomplete="off"
                    class="outline-none	py-3.5 px-3.5 w-full text-sm text-custom-black dark:text-custom-white bg-transparent bg-transparent text-h4 "
                    data-testid="private-key-input"
                    id="private-key"
                    name="privateKey"
                    placeholder="Enter your private key"
                    type="password"
                    value=""
                  />
                  <div
                    class="pr-3.5"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="dark:stroke-custom-white stroke-custom-black w-6"
                      data-aid="privateKeyShowHide"
                      tabindex="-1"
                      type="button"
                    >
                      <eye-off.svg
                        classname="w-5 h-5"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class=" mt-2 text-custom-grey100 dark:text-custom-white40 text-sm font-extrabold"
              >
                Account.enterPassword
              </div>
              <div
                class="mt-4 mb-8"
              >
                <div
                  class="flex relative rounded-md items-center hover:shadow-md w-full bg-input-light-bg dark:bg-custom-white10"
                >
                  <input
                    autocomplete="off"
                    class="outline-none	py-3.5 px-3.5 w-full text-sm text-custom-black dark:text-custom-white bg-transparent bg-transparent text-h4 "
                    data-testid="password"
                    id="enter-password"
                    name="password"
                    placeholder="Account.enterPassword"
                    type="password"
                    value=""
                  />
                  <div
                    class="pr-3.5"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="dark:stroke-custom-white stroke-custom-black w-6"
                      data-aid="passwordShowHide"
                      tabindex="-1"
                      type="button"
                    >
                      <eye-off.svg
                        classname="w-5 h-5"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="mt-2 ml-1 dark:text-white text-custom-grey100 font-normal text-[12px]"
                >
                  Min 8 characters
                </div>
              </div>
              <div
                class="flex mt-4 gap-2"
              >
                <button
                  class="flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full h-[44px] border-[1px] border-solid text-primary dark:text-custom-white border-primary dark:border-custom-white40 dark:hover:border-custom-white hover:border-primary-alt hover:text-primary-alt"
                  type="button"
                >
                  Actions.cancel
                </button>
                <button
                  class="flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full disabled:text-custom-grey40 dark:disabled:text-custom-grey40 h-[44px] text-custom-white dark:text-custom-white bg-custom-grey10"
                  data-testid="import-privatekey-btn"
                  disabled=""
                  type="submit"
                >
                  Actions.import
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dark:bg-surface-dark bg-custom-white flex flex-col h-[600px] dark:text-custom-white text-custom-black"
    >
      <div
        class="flex flex-row relative justify-between items-center h-16"
      >
        <button
          class="h-[24px] w-[24px] p-1 ml-1 hover:bg-custom-grey10 dark:hover:bg-custom-white10 rounded-full flex items-center justify-center absolute"
          type="button"
        >
          <arrow-left.svg
            classname="text-xl stroke-custom-black dark:stroke-custom-white"
          />
        </button>
        <div
          class="flex-1 text-center"
        >
          <div
            class="text-custom-black dark:text-custom-white text-lg font-extrabold"
          >
            Import account
          </div>
        </div>
      </div>
      <div
        class="w-full h-full overflow-auto"
      >
        <form>
           
          <div
            class="css-1vn5bei"
          >
            <div
              class="text-custom-black dark:text-custom-white text-sm font-regular"
            >
              Account.importAccountMessage
            </div>
            <div
              class="MuiBox-root css-1jrf70h"
            >
              <div
                class="flex relative rounded-md items-center hover:shadow-md w-full bg-input-light-bg dark:bg-custom-white10"
              >
                <input
                  autocomplete="off"
                  class="outline-none	py-3.5 px-3.5 w-full text-sm text-custom-black dark:text-custom-white bg-transparent bg-transparent text-h4 "
                  disabled=""
                  name="test"
                  placeholder="Private Key"
                  value="Private Key"
                />
              </div>
              <div
                class="relative flex flex-col items-center group"
              >
                <div
                  class="css-mzan16"
                >
                  <svg
                    class="inline-block select-none shrink-0 m-auto fill-custom-black dark:fill-custom-white text-2xl"
                    style="height: 1em; width: 1em;"
                  >
                    <help-circle.svg />
                  </svg>
                </div>
                <div
                  class="absolute bottom-0 flex flex-col items-center hidden mb-6 group-hover:flex right-0 undefined"
                >
                  <span
                    class="rounded-md dark:bg-surface-dark-alt bg-surface-light relative z-10 p-2 text-xs leading-none dark:text-custom-white text-custom-grey100 whitespace-no-wrap bg-black shadow-lg w-max max-w-[300px] mb-2"
                  >
                    Private Key
                  </span>
                  <div
                    class="w-3 h-3 -mt-2 rotate-45 bg-black"
                  />
                </div>
              </div>
            </div>
            <div
              class="mt-4 mb-8"
            >
              <div
                class="flex relative rounded-md items-center hover:shadow-md w-full bg-input-light-bg dark:bg-custom-white10"
              >
                <input
                  autocomplete="off"
                  class="outline-none	py-3.5 px-3.5 w-full text-sm text-custom-black dark:text-custom-white bg-transparent bg-transparent text-h4 "
                  data-testid="private-key-input"
                  id="private-key"
                  name="privateKey"
                  placeholder="Enter your private key"
                  type="password"
                  value=""
                />
                <div
                  class="pr-3.5"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="dark:stroke-custom-white stroke-custom-black w-6"
                    data-aid="privateKeyShowHide"
                    tabindex="-1"
                    type="button"
                  >
                    <eye-off.svg
                      classname="w-5 h-5"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class=" mt-2 text-custom-grey100 dark:text-custom-white40 text-sm font-extrabold"
            >
              Account.enterPassword
            </div>
            <div
              class="mt-4 mb-8"
            >
              <div
                class="flex relative rounded-md items-center hover:shadow-md w-full bg-input-light-bg dark:bg-custom-white10"
              >
                <input
                  autocomplete="off"
                  class="outline-none	py-3.5 px-3.5 w-full text-sm text-custom-black dark:text-custom-white bg-transparent bg-transparent text-h4 "
                  data-testid="password"
                  id="enter-password"
                  name="password"
                  placeholder="Account.enterPassword"
                  type="password"
                  value=""
                />
                <div
                  class="pr-3.5"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="dark:stroke-custom-white stroke-custom-black w-6"
                    data-aid="passwordShowHide"
                    tabindex="-1"
                    type="button"
                  >
                    <eye-off.svg
                      classname="w-5 h-5"
                    />
                  </button>
                </div>
              </div>
              <div
                class="mt-2 ml-1 dark:text-white text-custom-grey100 font-normal text-[12px]"
              >
                Min 8 characters
              </div>
            </div>
            <div
              class="flex mt-4 gap-2"
            >
              <button
                class="flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full h-[44px] border-[1px] border-solid text-primary dark:text-custom-white border-primary dark:border-custom-white40 dark:hover:border-custom-white hover:border-primary-alt hover:text-primary-alt"
                type="button"
              >
                Actions.cancel
              </button>
              <button
                class="flex items-center justify-center rounded-md hover:cursor-pointer font-extrabold text-sm w-full disabled:text-custom-grey40 dark:disabled:text-custom-grey40 h-[44px] text-custom-white dark:text-custom-white bg-custom-grey10"
                data-testid="import-privatekey-btn"
                disabled=""
                type="submit"
              >
                Actions.import
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
